src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py

```diff
@@ -126,12 +153,37 @@ def expand_parameters(self, para):
         return ret_para
 
     def update_search_space(self, search_space):
-        '''
-        Check if the search space is valid and expand it: support only 'choice', 'quniform', randint'
-        '''
-        self.expanded_search_space = self.json2parameter(search_space)
+        """
+        Check if the search space is valid and expand it: support only ``choice``, ``quniform``, ``randint``.
+
+        Parameters
+        ----------
+        search_space: dict
+            the format could be referred to [search space spec](https://nni.readthedocs.io/en/latest/Tutorial/SearchSpaceSpec.html).
+        """
+        self.expanded_search_space = self._json2parameter(search_space)
 
     def generate_parameters(self, parameter_id, **kwargs):
+        """
+        Generate parameters for one trial.
+
+        Parameters
+        ----------
+        parameter_id: int
+            the id for the generated hyperparameter
+        **kwargs
+            not used
+
+        Returns
+        -------
+        : dict
+            one configuration from the expanded search space. If all the configurations has been sent,
+            raise ``nni.NoMoreTrialError``.
```

Comment:
There can be a "raises" section. https://numpydoc.readthedocs.io/en/latest/format.html
Also I don't think there need to be a explicit see also section for `NoMoreTrialError`.

Reply:
fixed, thanks